doc/Language/classtut.pod6

```diff
@@ -484,12 +484,96 @@ X<|FQN>
 
 Just like attributes, methods can also be private. Private methods are declared
 with a prefixed exclamation mark. They are called with C<self!>
-followed by the method's name. To call a private method of another class the
-calling class has to be trusted by the called class. A trust relationship is
-declared with C<trusts> and the class to be trusted must already be declared.
-Calling a private method of another class requires an instance of that class
-and the fully qualified name (FQN) of the method. Trust also allows access to
-private attributes.
+followed by the method's name. In the following implementation of a C<Region>
+class, methods C<population-info> and C<densely-populated> are private methods,
+while the rest are public ones:
+
+=begin code
+class Region {
+    has $.name;
+    has $.population;
+    has $.area-size;
+    has $.continent;
+    has $.high-population-density;
+
+    method info {
+        say "Name: $!name";
+        say self!population-info;
+    }
+
+    method more-densely-populated(Region:D $r) {
+        my $result = self.population-density > $r.population-density ?? 'more' !! 'less';
+        say "!name is $result densely populated than $r..name.";
+    }
+
+    method population-density {
+        ($!population / $!area-size).round(0.01)
+    }
+
+    method can-be-crowded {
+        self!densely-populated
+        ?? "$!name can be crowdy."
+        !! "There is enough space in $!name."
+    }
+
+    method !population-info(--> Str:D) {
+        "$!population people live here."
+    }
+
+    method !densely-populated(--> Bool:D) {
+        self.population-density > $!high-population-density
+    }
+}
+
```

Comment:
First, thanks a lot for your contribution. However, there are a couple of things that could be improved here.
1. Methods shouldn't do I/O unless explicitly requested to do so. It's probably better to return a string, and let the client decide what to do with them, than "say" that same string.
2. There are probably better use cases than the one here, which seems to be "these methods are private because I want to illustrate privacy of methods". Both could be private, or not, and it would probably make more sense for them to be public. Documentation should reflect intent #1748, not syntax. So it would be probably better to create an example where it makes sense for them to be private.
3. You should probably also establish some identifier convention, like preceding them with a `_`.
4. Don't particularly like also method returns as strings; you should use the most adequate type for it, or define an enum. `can-be-crowded` should return a `Bool`, for instance, and again let the client decide what to do with that info.